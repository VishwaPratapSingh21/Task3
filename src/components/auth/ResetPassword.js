import React, { Component } from 'react';
import {notify} from 'react-notify-toast';
import  axiosInstance from "../Apicalls";

/**
 * Component for resetting password.
 * @author [Hadijah kyampeire](https://github.com/hadijahkyampeire/Yummy_Reactjs_frontend)
 */
class ResetPassword extends Component {
    state = {
        email:'',
        password:'',
        retyped_password:'',
        token: null,

    }
    // called when the component has been rendered
    componentDidMount = () => {
        const search = this.props.location.search;
        const token = decodeURIComponent(search.substring(4, search.length));
        this.setState({ token })
    };

    handleInputChange =(event) =>{
        const {name, value } =event.target;
        this.setState({[name]:value});
    }

    handleResetPassword = (e)=>{
      // prevent the default action of the event to be triggered
        e.preventDefault();
        const headers = {Authorization: `Bearer ${this.state.token}`}
        const {email, password, retyped_password} =this.state; 
        axiosInstance.put(`auth/reset_password`, {email, password, retyped_password},{headers})
        .then(response=>{
            // success message once the Api call is successful
            notify.show(response.data.message, 'success', 4000)
            this.props.history.push('/login')
        }).catch(error=>{
            if (error.response){
              // notify about errors due to response
                notify.show(error.response.data.message, 'error', 3000)
            }else if(error.request){
              // error message due to request
                notify.show("Request not made", 'error', 3000)
            }
        })
    }
    
  render() {
    return (
      <div className="background">
      <div id="changepassword">

      <div id="form-group">
      <h3>Password Reset</h3>
        <form onSubmit={this.handleResetPassword}>
          <div className="input-group">
            <span className="input-group-addon pr-4" id="email"><i className="fa fa-envelope" /></span>
            <input
              type="email"
              id="emailsignup"
              name="email"
              value={this.state.email}
              onChange={this.handleInputChange}
              aria-describedby="email"
              className="form-control"
              placeholder="example@gmail.com"
            />
          </div>
          <br />
          <div className="input-group">
            <span className="input-group-addon pr-4" id="email"><i className="fa fa-lock" /></span>
            <input
              type="password"
              name="password"
              className="form-control"
              value={this.state.password}
              onChange={this.handleInputChange}
              placeholder="Password"
            />
          </div>
          <br />
          <div className="input-group">
            <span className="input-group-addon pr-4" id="email"><i className="fa fa-lock" /></span>
            <input
              type="password"
              name="retyped_password"
              className="form-control"
              value={this.state.retyped_password}
              onChange={this.handleInputChange}
              placeholder="retype_Password"
            />
          </div>
          <br />

          <button name="signin" className="btn btn-default" id="signin" type="submit">
            <i className="glyphicon glyphicon-log-in"> Save</i>
          </button>
        </form>
        </div>
        </div>
      </div>

    );
  }
}
export default ResetPassword;
