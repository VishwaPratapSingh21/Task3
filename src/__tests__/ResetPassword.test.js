import React from 'react';
import { shallow, enzyme } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import ResetPassword from '../components/auth/ResetPassword';

describe('ResetPassword component', () => {
  const wrapper = shallow(<ResetPassword location={{ search: 'testUrl' }} />);
  const preventDefault = jest.fn();
  

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders initial states', () => {
    expect(wrapper.state().email).toEqual('');
    expect(wrapper.state().password).toEqual('');
    expect(wrapper.state().retyped_password).toEqual('');
  });

  it('renders a form', () => {
    expect(wrapper.find('form')).toHaveLength(1);
    expect(wrapper.find('form').simulate('submit', {preventDefault}))
    expect(preventDefault).toBeCalled();
    expect(wrapper.find('[name="email"]').simulate('change', {target:{name:'email', value:'had@hmail.com'}}))
  });

  it('renders a button', () => {
    expect(wrapper.find('button')).toHaveLength(1);
  });

  it('has divs', () => {
    expect(wrapper.find('div')).toHaveLength(6);
  });

  it('has span', () => {
    expect(wrapper.find('span')).toHaveLength(3);
  });

  it('has icon h3', () => {
    expect(wrapper.find('h3')).toHaveLength(1);
  });

  it('has inputs', () => {
    expect(wrapper.find('input')).toHaveLength(3);
  });

  it('has br', () => {
    expect(wrapper.find('br')).toHaveLength(3);
  });
});
