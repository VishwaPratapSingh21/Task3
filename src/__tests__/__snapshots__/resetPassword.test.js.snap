// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword component renders properly 1`] = `
<div
  className="background"
>
  <div
    id="changepassword"
  >
    <div
      id="form-group"
    >
      <h3>
        Password Reset
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-group"
        >
          <span
            className="input-group-addon pr-4"
            id="email"
          >
            <i
              className="fa fa-envelope"
            />
          </span>
          <input
            aria-describedby="email"
            className="form-control"
            id="emailsignup"
            name="email"
            onChange={[Function]}
            placeholder="example@gmail.com"
            type="email"
            value=""
          />
        </div>
        <br />
        <div
          className="input-group"
        >
          <span
            className="input-group-addon pr-4"
            id="email"
          >
            <i
              className="fa fa-lock"
            />
          </span>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <br />
        <div
          className="input-group"
        >
          <span
            className="input-group-addon pr-4"
            id="email"
          >
            <i
              className="fa fa-lock"
            />
          </span>
          <input
            className="form-control"
            name="retyped_password"
            onChange={[Function]}
            placeholder="retype_Password"
            type="password"
            value=""
          />
        </div>
        <br />
        <button
          className="btn btn-default"
          id="signin"
          name="signin"
          type="submit"
        >
          <i
            className="glyphicon glyphicon-log-in"
          >
             Save
          </i>
        </button>
      </form>
    </div>
  </div>
</div>
`;
